apps/dashboard/src/components/workflow-editor/steps/configure-step-form.tsx

```diff
@@ -23,70 +24,137 @@ import { Input, InputField } from '@/components/primitives/input';
+
+const STEP_TYPE_TO_CONTROL_VALUES_FORM: Record<StepTypeEnum, () => React.JSX.Element | null> = {
+  [StepTypeEnum.DELAY]: DelayControlValues,
+  [StepTypeEnum.IN_APP]: () => null,
+  [StepTypeEnum.EMAIL]: () => null,
+  [StepTypeEnum.SMS]: () => null,
+  [StepTypeEnum.CHAT]: () => null,
+  [StepTypeEnum.PUSH]: () => null,
+  [StepTypeEnum.CUSTOM]: () => null,
+  [StepTypeEnum.TRIGGER]: () => null,
+  [StepTypeEnum.DIGEST]: () => null,
+};
+
+const calculateDefaultControlsValues = (step: StepDataDto) => {
+  if (Object.keys(step.controls.uiSchema ?? {}).length !== 0) {
+    return merge(buildDefaultValues(step.controls.uiSchema ?? {}), step.controls.values);
+  }
 
-const SUPPORTED_STEP_TYPES = [StepTypeEnum.IN_APP];
+  return merge(buildDefaultValuesOfDataSchema(step.controls.dataSchema ?? {}), step.controls.values);
+};
 
 type ConfigureStepFormProps = {
   workflow: WorkflowResponseDto;
   environment: IEnvironment;
   step: StepDataDto;
+  issues?: StepIssuesDto;
   update: (data: UpdateWorkflowDto) => void;
   updateStepCache: (step: Partial<StepDataDto>) => void;
 };
 
 export const ConfigureStepForm = (props: ConfigureStepFormProps) => {
-  const { step, workflow, update, updateStepCache, environment } = props;
-  const navigate = useNavigate();
-  const isCodeCreatedWorkflow = workflow.origin === WorkflowOriginEnum.EXTERNAL;
+  const { step, workflow, update, updateStepCache, environment, issues } = props;
+
+  const isSupportedStep = !UNSUPPORTED_STEP_TYPES.includes(step.type);
+  const isReadOnly = !isSupportedStep || workflow.origin === WorkflowOriginEnum.EXTERNAL;
+
+  const isTemplateConfigurableStep = isSupportedStep && TEMPLATE_CONFIGURABLE_STEP_TYPES.includes(step.type);
+  const isInlineConfigurableStep = isSupportedStep && INLINE_CONFIGURABLE_STEP_TYPES.includes(step.type);
 
+  const navigate = useNavigate();
   const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
 
   const onDeleteStep = () => {
     update({ ...workflow, steps: workflow.steps.filter((s) => s._id !== step._id) });
     navigate(buildRoute(ROUTES.EDIT_WORKFLOW, { environmentSlug: environment.slug!, workflowSlug: workflow.slug }));
   };
 
-  const defaultValues = {
-    name: step.name,
-    stepId: step.stepId,
-  };
+  const registerInlineControlValues = useMemo(() => {
+    return (step: StepDataDto) => {
+      if (isInlineConfigurableStep) {
+        return {
+          controlValues: calculateDefaultControlsValues(step),
+        };
+      }
+
+      return {};
+    };
+  }, [isInlineConfigurableStep]);
+
+  const controlsSchema = useMemo(
+    () => (isInlineConfigurableStep ? buildDynamicZodSchema(step.controls.dataSchema ?? {}) : undefined),
+    [step.controls.dataSchema, isInlineConfigurableStep]
+  );
+
+  const defaultValues = useMemo(
+    () => ({
+      name: step.name,
+      stepId: step.stepId,
+      ...registerInlineControlValues(step),
+    }),
+    [step, registerInlineControlValues]
+  );
```

Comment:
We need to register control values and controls schema here conditionally. I thought maybe it could be done in a separate component like:

```
const form = useFormContext()

form.register(`controlValues`, registerInlineControlValues)
form.resolver = controlsSchema() // not possible
....
```
But thats not feasible because:
1.  the resolver can't be changed/extended
2. the `controlValues` cant be registered conditionally because then the `useFormAutosave` -> `data` object must have some type, unless forced to `any`

If I removed the `isInlineConfigurableStep` condition altogether, then if try to change just name on In-App step, it would trigger validation on `controlValues` as well.